### DISCUSSION TITLE
Best approach on statically building a placeholder dashboard, and hydrate client side?

### DISCUSSION THREAD

**Discussion by mgm1313** [vercel/next.js#12043] [Help]
Hi folks,

Basically what I want to do is build a dashboard. I want to statically generate the placeholder content, and then on the client side hydrate it with actual data from a graphql server. Eventually authentication will be added. (So basically the Zeit dashboard approach, if I'm correct). 

At the moment I'm playing with [SWR](https://github.com/zeit/swr) which seems to do the job. But of course I also want to do mutations, and a lot of articles out there suggest Apollo is a better fit for that task. 
So I started looking through the Apollo examples, but I believe none of them are up to date yet with the new getStaticProps/getServerSideProps (https://github.com/zeit/next.js/issues/11014). 

So at the moment I'm wondering 2 things: 
1. is Apollo indeed the best (or let's say, better than SWR) tool for the job, and why? 
2. If I where to use Apollo, what would be the approach to succeed in this goal of statically serving a placeholder and hydrating on client side?

If somebody could point me to, or give, some reading material that would be great!

**Comment by leerob** [comment]
[SWR](https://github.com/zeit/swr#api) has a `mutate` API that should achieve what you're looking for. 

https://paco.im/blog/shared-hook-state-with-swr
